plugins/e57/io/E57Reader.cpp

```diff
@@ -1,228 +1,127 @@
-#include <E57Exception.h>
+ * Copyright (c) 2019, Helix Re Inc.
+ *
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following
+ * conditions are met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in
+ *       the documentation and/or other materials provided
+ *       with the distribution.
+ *     * Neither the name of Helix Re Inc. nor the
+ *       names of its contributors may be used to endorse or promote
+ *       products derived from this software without specific prior
+ *       written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT
+ * OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY
+ * OF SUCH DAMAGE.
+ ****************************************************************************/
 
 #include "E57Reader.hpp"
 #include "Utils.hpp"
+#include "arbiter/arbiter.hpp"
 
 namespace pdal
 {
 
-E57Reader::ChunkReader::ChunkReader(const point_count_t &pointOffset,
-    const point_count_t &maxPointRead, const std::shared_ptr<e57::Scan>& scan,
-    const std::set<std::string>& e57Dimensions) :
-    m_startIndex(0), m_maxPointRead(maxPointRead),
-    m_defaultChunkSize(1000000), m_scan(scan)
-{
-    // Initialise the read buffers
-    for (auto& dimension: e57Dimensions)
-        m_doubleBuffers[dimension] = std::vector<double>(m_defaultChunkSize,0);
-
-    auto vectorNode = scan->getPoints();
-    auto prototype = static_cast<e57::StructureNode>(vectorNode.prototype());
-    for (auto& keyValue: m_doubleBuffers)
-        m_e57buffers.emplace_back(vectorNode.destImageFile(), keyValue.first,
-             keyValue.second.data(), m_defaultChunkSize, true,
-             (prototype.get(keyValue.first).type() == e57::E57_SCALED_INTEGER));
-
-    m_pointOffset = pointOffset;
-
-    // Setup the reader
-    m_dataReader = std::unique_ptr<e57::CompressedVectorReader>(
-        new e57::CompressedVectorReader(vectorNode.reader(m_e57buffers)));
-}
+static PluginInfo const s_info{"readers.e57", "Reader for E57 files", ""};
```

Comment:
This should contain a link to the stage documentation.

Reply:
done!!